flow/activities/cancel_table_addition_activity.go

```diff
@@ -0,0 +1,243 @@
+package activities
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/PeerDB-io/peerdb/flow/alerting"
+	"github.com/PeerDB-io/peerdb/flow/generated/protos"
+	"github.com/PeerDB-io/peerdb/flow/internal"
+	"github.com/PeerDB-io/peerdb/flow/otel_metrics"
+	"github.com/PeerDB-io/peerdb/flow/shared"
+	"github.com/jackc/pgx/v5"
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+	"google.golang.org/protobuf/proto"
+)
+
+type CancelTableAdditionActivity struct {
+	CatalogPool    shared.CatalogPool
+	Alerter        *alerting.Alerter
+	TemporalClient client.Client
+	OtelManager    *otel_metrics.OtelManager
+}
+
+func (a *CancelTableAdditionActivity) GetCompletedTablesInQrepRuns(ctx context.Context, flowJobName string) ([]string, error) {
+	shutdown := heartbeatRoutine(ctx, func() string {
+		return "fetching completed tables from qrep_runs"
+	})
+	defer shutdown()
+	rows, err := a.CatalogPool.Query(ctx, "SELECT source_table FROM peerdb_stats.qrep_runs WHERE parent_mirror_name = $1 AND consolidate_complete = true", flowJobName)
```

Comment:
Question, what happens when a table was added before, completed snapshot, then removed, and now added and stuck? Will there be a consolidate_complete=true for it?